### DISCUSSION TITLE
How to Make Scrollable Areas "pan with Touch or Mouse Drag"?

### DISCUSSION THREAD

**Discussion by najak3d** [AvaloniaUI/Avalonia#7474] [Q&A] [answered]
On Mobile devices, it is expected that dragging the screen content scrolls the view (and even has momentum, allowing you to flick-throw the content, which continues and decelerates after the flick).

Is this type of behavior already planned for this year as a "base behavior option" of Avalonia?  Or will we need to make some sort of Custom Behavior to make it work like this?

**Comment by kekekeks** [comment, marked as answer]
ScrollViewer enables _touch_-based scrolling out of the box. You can enable it for custom controls by adding `ScrollGestureRecognizer` like this:

https://github.com/AvaloniaUI/Avalonia/blob/fed63630124dcdf4a2fcc10b1f9d0ca822d568f5/src/Avalonia.Themes.Fluent/Controls/ScrollViewer.xaml#L39-L43

**Reply by najak3d** [reply]
That's Perfect, thanks!   Avalonia wins the day again. (and again, and again!)

**Reply by nettlep** [reply]
Can you confirm if this should also work on Desktop applications? I'm building for Raspberry Pi with a touchscreen (no keyboard or mouse) and find the mouse drag does not work. For my case, I must touch the scrollbar, wait for it to expand, then drag the scrollbar. This makes for a rather quirky and awkward user experience.

**Reply by thevortexcloud** [reply]
Some of the gestures are touch only (meaning they will work on a desktop, as long as it has a  touch screen). Still not clear why. Not sure if the scroll one is one of those though.

https://github.com/AvaloniaUI/Avalonia/discussions/17493

**Reply by nettlep** [reply]
I may not have been clear. I'm using a touchscreen and the touch gestures are not working. I'm wondering if it is because it is a desktop application (runs on raspberry pi using the classic desktop lifetime.)

In my case, it is a standalone device with a touchscreen as the only IO device (no keyboard or mouse.)

**Comment by marcussacana** [comment]
Just a note to anyone reading this too.
I created a ScrollViewner with buttons inside, and the button does not allow the propagation of the touch event, I managed to solve it by making a new class to extends the button
```csharp
public class TapScrollableButton : Button, IStyleable
{
    Type IStyleable.StyleKey => typeof(Button);
    protected override void OnPointerPressed(PointerPressedEventArgs e)
    {
        base.OnPointerPressed(e);

        //Allow ScrollViewer handle touch scroll 
        e.Handled = false;
    }
}
```
If anyone knows better implementation for my case, just tell me, I'm still new to avalonia.
